tests/components/bayesian/test_config_flow.py

```diff
@@ -0,0 +1,1540 @@
+            },
+        )
+        await hass.async_block_till_done()
+
+        # Since "add_another" was True, we should return to the observation selector
+        assert result3["step_id"] == str(ConfigFlowSteps.OBSERVATION_SELECTOR)
+        assert result3["type"] is FlowResultType.MENU
+
+        # Add a state observation
+        result4 = await hass.config_entries.flow.async_configure(
+            result3["flow_id"], {"next_step_id": str(ObservationTypes.STATE)}
+        )
+        await hass.async_block_till_done()
+
+        assert result4["step_id"] == str(ObservationTypes.STATE)
+        assert result4["type"] is FlowResultType.FORM
+        result5 = await hass.config_entries.flow.async_configure(
+            result4["flow_id"],
+            {
+                CONF_ENTITY_ID: "sensor.work_laptop",
+                CONF_TO_STATE: "on",
+                CONF_P_GIVEN_T: 60,
+                CONF_P_GIVEN_F: 20,
+                CONF_NAME: "Work laptop on network",
+                "add_another": True,
+            },
+        )
+        await hass.async_block_till_done()
+
+        # Since "add_another" was True, we should return to the observation selector
+        assert result5["step_id"] == str(ConfigFlowSteps.OBSERVATION_SELECTOR)
+        assert result5["type"] is FlowResultType.MENU
+
+        # Lastly, add a template observation
+        result6 = await hass.config_entries.flow.async_configure(
+            result5["flow_id"], {"next_step_id": str(ObservationTypes.TEMPLATE)}
+        )
+        await hass.async_block_till_done()
+
+        assert result6["step_id"] == str(ObservationTypes.TEMPLATE)
+        assert result6["type"] is FlowResultType.FORM
+        result7 = await hass.config_entries.flow.async_configure(
+            result6["flow_id"],
+            {
+                CONF_VALUE_TEMPLATE: """
+{% set current_time = now().time() %}
+{% set start_time = strptime("07:00", "%H:%M").time() %}
+{% set end_time = strptime("18:30", "%H:%M").time() %}
+{% if start_time <= current_time <= end_time %}
+True
+{% else %}
+False
+{% endif %}
+                """,
+                CONF_P_GIVEN_T: 45,
+                CONF_P_GIVEN_F: 5,
+                CONF_NAME: "Daylight hours",
+                "add_another": False,
+            },
+        )
+
+        # Since add_another is false, we should now be done.
+        assert result7["version"] == 1
+        assert result7["options"] == {
+            CONF_NAME: "Office occupied",
+            CONF_PROBABILITY_THRESHOLD: 0.5,
+            CONF_PRIOR: 0.15,
+            CONF_DEVICE_CLASS: "occupancy",
+            CONF_OBSERVATIONS: [
+                {
+                    CONF_PLATFORM: str(ObservationTypes.NUMERIC_STATE),
+                    CONF_ENTITY_ID: "sensor.office_illuminance_lux",
+                    CONF_ABOVE: 40,
+                    CONF_P_GIVEN_T: 0.85,
+                    CONF_P_GIVEN_F: 0.45,
+                    CONF_NAME: "Office is bright",
```

Comment:
I don't think there should be names here as earlier mentioned

Reply:
I'm not sure which earlier mention you are referring to? `CONF_NAME`s are required as you need something user-readable to label each observation so that they can be identified selected for editing and deletion in the options flow.

Reply:
I get that and perhaps that's what we need to have (as I'm not that into how this operates). 
Otherwise maybe just `Observation 1; sensor.xyz` if it's not common you use the same input sensor for multiple observations.

Reply:
For numeric ranges, it will be quite common to have multiple entries for the same sensor.
Whats more for `template` observations I can't think of a way to elegantly auto-generate a title/name short of just using the whole template string which is of arbitrary length...